vscode/src/client/analyzerClient.ts

```diff
@@ -50,6 +51,11 @@ export class AnalyzerClient {
     this.analyzerServer.on("exit", (code) => {
       this.outputChannel.appendLine(`Analyzer exited with code ${code}`);
     });
+
+    const enableGenerativeAi = this.config!.get("enableGenerativeAI") as boolean;
```

Comment:
Suggest you pull this out into a `isGenAiEnabled` func we can call in other places as we need it.